site-builder/src/main.rs

```diff
@@ -242,40 +218,52 @@ async fn run() -> Result<()> {
 
     match &args.command {
         Commands::Publish {
-            directory,
-            content_encoding,
-            site_name,
-            epochs,
-            list_directory,
+            publish_options:
+                PublishOptions {
+                    directory,
+                    content_encoding,
+                    site_name,
+                    epochs,
+                    list_directory,
+                },
         } => {
             publish_site(
-                directory,
-                content_encoding,
-                site_name,
+                PublishOptions {
+                    directory: directory.to_path_buf(),
+                    content_encoding: *content_encoding,
+                    site_name: site_name.to_string(),
+                    epochs: *epochs,
+                    list_directory: *list_directory,
+                },
                 &config,
-                *epochs,
-                *list_directory,
             )
             .await?
         }
```

Comment:
Suggestion: Instead of destructuring and reconstructing the `PublishOptions`, let's do one of the following:
1. Change `match &args.command` to `match args.command` and then use `publish_options` directly:
   ```suggestion
           Commands::Publish { publish_options } => publish_site(publish_options, &config).await?,

   ```
   This requires changing some lines below (removing some `*`, `to_string`, ..., and adding some `&`).
2. Add a `#[derive(Clone)]` to `PublishOptions` (which we should anyway have) and change this to
   ```rust
           Commands::Publish { publish_options } => {
               publish_site(publish_options.clone(), &config).await?
           }
   ```

The same applies to the `Update` command below.

Reply:
Got it! Why do we need to use this though?
> Change match &args.command to match args.command

Reply:
The `publish_site` takes `publish_options` as an owned object; if we do `match &args.command` we only have it as a reference. Of course, an additional option would be to change the argument of `publish_site` to `publish_options: &PublishOptions`.

Reply:
Ah, I just realized that my suggestion above was a bit unclear. I meant to do one of the two things, not both (although both together also works). 😇